packages/components/checkbox/src/composables/use-checkbox-status.ts

```diff
@@ -0,0 +1,48 @@
+import { computed, inject, ref, toRaw } from 'vue'
+import { toTypeString } from '@vue/shared'
+import { useSize } from '@element-plus/hooks'
+import { checkboxGroupContextKey } from '@element-plus/tokens/checkbox'
+
+import type { ComponentInternalInstance } from 'vue'
+import type { CheckboxProps } from '../checkbox'
+import type { CheckboxModel } from '../composables'
+
+export const useCheckboxStatus = (
+  props: CheckboxProps,
+  slots: ComponentInternalInstance['slots'],
+  { model }: Partial<CheckboxModel>
+) => {
+  const checkboxGroup = inject(checkboxGroupContextKey, undefined)
+  const isFocused = ref(false)
+  const isChecked = computed<boolean>(() => {
+    const value = model!.value
+    if (toTypeString(value) === '[object Boolean]') {
```

Comment:
```suggestion
    if (isBoolean(value)) {
```

Maybe use the `isBoolean` functions from @element-plus/utils